src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -337,16 +337,42 @@ public bool InitializeForBatch(TaskLoggingContext loggingContext, ItemBucket bat
             }
 
             string realTaskAssemblyLocation = TaskInstance.GetType().Assembly.Location;
```

Comment:
ugh I'll have to make a different code path here for [mt inline tasks](https://github.com/dotnet/msbuild/pull/11948), because those tasks won't have location

Reply:
no action for you just noting down my thought process, feel free to resolve